tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -1200,11 +1200,13 @@ class SerializableModelLibraryGenerator {
     List<SerializableModelFieldDefinition> fields,
   ) {
     var map = fields.fold<Map<Code, Expression>>({}, (map, field) {
+      var fieldName = _createSerializableFieldNameReference(serverCode, field);
+
       return {
         ...map,
         if (field.type.nullable)
-          Code("if (${field.name} != null) '${field.name}'"): refer(field.name),
-        if (!field.type.nullable) Code("'${field.name}'"): refer(field.name),
+          Code("if (${fieldName.symbol} != null) '${field.name}'"): fieldName,
```

Comment:
Question: What exactly is fixed here? Since you haven't regenerated the project it seems like this would produce the same output as before.

Reply:
This fixes the implicit field case, hence the call to the method `_createSerializableFieldNameReference` that will append `_` incase the field is implicit. Without that any implicit relation breaks.

Reply:
After call on slack: The change here is probably correct. We identified a scenario that wasn't covered by over models yet so that will be created first and merged to main and then we will notice if this change has any impact.

The scenario that was missed was if a model was created with a table but that has a field with scope none.